packages/core/src/PhysicsManager.ts

```diff
@@ -0,0 +1,121 @@
+import { Ray, Vector3 } from "@oasis-engine/math";
+import { Layer } from "./Layer";
+import { ColliderFeature } from "./collider";
+import { Engine } from "./Engine";
+import { Collider } from "./collider/Collider";
+
+/**
+ * Structure used to get information back from a raycast or a sweep.
+ */
+export class HitResult {
+  /** The collider that was hit. */
+  collider: Collider = null;
+  /** The distance from the origin to the hit point. */
+  distance: Number = Number.MAX_VALUE;
+  /** The hit point of the collider that was hit in world space. */
+  point: Vector3 = new Vector3();
+  /** The hit normal of the collider that was hit in world space. */
+  normal: Vector3 = new Vector3();
+
+  reInit(distance: number = Number.MAX_VALUE) {
```

Comment:
This is a internal func, you must named "_ reInit()" and add @internal comments.